src/Model/GroundWaterFlow/gwf3maw8.f90

```diff
@@ -3322,59 +3326,78 @@ subroutine maw_calculate_satcond(this, i, j, node)
       end if
     end if
     !
+    ! -- calculate the aquifer transmissivity (T2pi)
+    T2pi = DTWOPI * tthka * sqrtk11k22
+    !
     ! -- calculate effective radius
     if (this%dis%ndim == 3 .and. this%ieffradopt /= 0) then
       Txx = k11 * tthka
       Tyy = k22 * tthka
       dx = sqrt(this%dis%area(node))
       dy = dx
-      yx4 = (Tyy/Txx)**0.25D0
-      xy4 = (Txx/Tyy)**0.25D0
-      eradius = 0.28D0 *((yx4*dx)**2 +(xy4*dy)**2)**0.5D0 / (yx4+xy4)
+      yx4 = (Tyy/Txx)**DQUARTER
+      xy4 = (Txx/Tyy)**DQUARTER
+      eradius = 0.28_DP * ((yx4*dx)**DTWO + (xy4*dy)**DTWO)**DHALF / (yx4+xy4)
     else
       area = this%dis%area(node)
       eradius = sqrt(area / (DEIGHT * DPI))
     end if
     !
-    ! -- conductance calculated using Thiem equation
-    if (this%mawwells(i)%ieqn == 1) then
-      c = (DTWO * DPI * tthka * sqrtk11k22) / log(eradius / this%mawwells(i)%radius)
-    ! -- conductance calculated using skin
-    else if (this%mawwells(i)%ieqn == 2) then
-      hks = this%mawwells(i)%hk(j)
-      ! prevent division by zero
-      if (tthkw * hks > DZERO) then
-        skin = (((sqrtk11k22*tthka)/(hks*tthkw)) - DONE) * &
-               log(this%mawwells(i)%sradius(j)/this%mawwells(i)%radius)
-        c = (DTWO * DPI * tthka * sqrtk11k22) / skin
-      end if
-    ! -- conductance calculated using cumulative Thiem and skin equations
-    else if (this%mawwells(i)%ieqn == 3) then
-      ! calculate lc1
-      lc1 = log(eradius / this%mawwells(i)%radius) / (DTWO * DPI * tthka * sqrtk11k22)
-      ! calculate lc2
+    ! -- conductance calculations
+    ! -- Thiem equation (1) and cumulative Thiem and skin equations (3)
+    if (this%mawwells(i)%ieqn == 1 .or. this%mawwells(i)%ieqn == 3) then
+      lc1 = log(eradius / this%mawwells(i)%radius) / T2pi
+    end if
+    ! -- skin equation (2) and cumulative Thiem and skin equations (3)
+    if (this%mawwells(i)%ieqn == 2 .or. this%mawwells(i)%ieqn == 3) then
       hks = this%mawwells(i)%hk(j)
-      ! prevent division by zero
       if (tthkw * hks > DZERO) then
-        skin = (((sqrtk11k22*tthka)/(hks*tthkw)) - DONE) * &
-               log(this%mawwells(i)%sradius(j)/this%mawwells(i)%radius)
-        lc2 = skin / (DTWO * DPI * tthka * sqrtk11k22)
-      else
-        lc2 = DZERO
-      end if
-      ! calculate conductance
-      denom = lc1 + lc2
-      if (denom > DZERO) then
-        c = DONE / denom
+        skin = (((sqrtk11k22 * tthka) / (hks * tthkw)) - DONE) *                 &

```

Comment:
Do we need a divide by zero check for the denominator here?  Or do we check and make sure that user-provided hks is greater than 0 in the read part.  What about tthkw?  Any way that could be zero?